nixos/modules/services/networking/kea.nix

```diff
@@ -9,20 +9,26 @@ with lib;
 let
   cfg = config.services.kea;
 
+  xor = x: y: (!x && y) || (x && !y);
   format = pkgs.formats.json {};
 
-  ctrlAgentConfig = format.generate "kea-ctrl-agent.conf" {
+  choose = x: y: if x != null then x else y;
```

Comment:
I'm not happy with this function name and I'm having a tough time finding a better one.

Reply:
Fine, renamed.